### PR TITLE
Home work 6

### PR DESCRIPTION
**Pull request by UnitOmega13**

### CODE REVIEW

File: src/Pair.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+		if (this == object) return true;
+		if (object == null || getClass() != object.getClass()) return false;
+		Pair<?, ?> pair = (Pair<?, ?>) object;
+		return Objects.equals(getFirst(), pair.getFirst()) &&
+				Objects.equals(getSecond(), pair.getSecond());
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(getFirst(), getSecond());
```
Comment (Sofasmile) [inline src/Pair.java]: think about other hashCode implementation

File: src/Storage.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import com.sun.istack.internal.Nullable;
+
+import java.util.HashMap;
+
+public class Storage<T, K> extends HashMap{
+
+	@Override
```
Comment (Sofasmile) [inline src/Storage.java]: create your own implementation

**Review by Sofasmile — COMMENTED** [review]
resolve comments

File: src/Storage.java (hunk 2 above)
Comment (UnitOmega13) [reply src/Storage.java]: I deleted extending and change logic and change hashCode in Pair.

File: src/Storage.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+			System.out.println("Sorry item does not exist!`");
+		}
+		return listOfValues.get(key - 1);
+	}
+
+
+	void put(int key, Object value) {
+		listOfKeys.add(key);
+		listOfValues.add(value);
+
```
Comment (Sofasmile) [inline src/Storage.java]: cs

File: src/Pair.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+	private K getSecond() {
+		return second;
+	}
+
+	public static <T, K> Pair<T, K> of(T first, K second) {
+		return new Pair<>(first, second);
+	}
+
+	@Override
+	public boolean equals(Object object) {
```
Comment (Sofasmile) [inline src/Pair.java]: what about your implementation?

File: src/Storage.java (hunk 3 above)
Comment (UnitOmega13) [reply src/Storage.java]: fixed

File: src/Pair.java (hunk 4 above)
Comment (UnitOmega13) [reply src/Pair.java]: I have implement equals

**Review by Sofasmile — APPROVED** [review]
good

**Pull request closed without merging**